### PR TITLE
let client-gen understand vendor

### PR DESCRIPTION
**Pull request by caesarxuchao**
This is extracted from #44784, where we move external api types to k8s.io/api. After the move, the types will locate at vendor/k8s.io/api/xxx. However, gengo index the parsed package using the import path, which is stripped of  the vendor/ prefix, so we'll need to strip the vendor/ prefix as necessary in client-gen.

This PR doesn't produce any change to the generated clientset yet since all types are still in the kubernetes repo.

### CODE REVIEW

File: cmd/libs/go2idl/client-gen/path/BUILD [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (mbohlool) [inline cmd/libs/go2idl/client-gen/path/BUILD]: Please separate generated and non-generated code.

File: cmd/libs/go2idl/client-gen/path/path.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package path
+
+import "strings"
+
+// Vendorless removes the longest match of "*/vendor/" from the front of p.
```
Comment (mbohlool) [inline cmd/libs/go2idl/client-gen/path/path.go]: can you add why that works, maybe with one example?

**Review by mbohlool — DISMISSED** [review]

File: cmd/libs/go2idl/client-gen/path/BUILD (hunk 1 above)
Comment (mbohlool) [inline cmd/libs/go2idl/client-gen/path/BUILD]: Sorry, ignore this.

**Comment by mbohlool** [comment]
one nit. otherwise LGTM.

File: cmd/libs/go2idl/client-gen/path/path.go (hunk 2 above)
Comment (caesarxuchao) [inline cmd/libs/go2idl/client-gen/path/path.go]: Added.

**Comment by caesarxuchao** [comment]
@k8s-bot kops aws e2e test this

File: hack/.linted_packages [hunk 3]
```diff
@@ -26,6 +26,7 @@ cmd/kubectl
 cmd/kubelet
 cmd/libs/go2idl/client-gen
 cmd/libs/go2idl/client-gen/generators
+cmd/libs/go2idl/client-gen/path
```
Comment (caesarxuchao) [inline hack/.linted_packages]: Eh, this is the only change to hack/ and I don't think it is worth bothering one of hack/OWNERS to approve it. I'll try to manually add the approve tag.

**Comment by caesarxuchao** [comment]
@mbohlool it seems lgtm and approve needs to be in two lines? I added the label manually :)

**Pull request merged**